### PR TITLE
Remove wl-unicode name of Tilde (also known as Infix)

### LINKED ISSUE #44: Tilde vs RawTilde
In the current database in master, there is a mistake: `a\[RawTilde]b\[RawTilde]c` should be parsed as `b[a,c]`,while
 'a\[Tilde]b\[Tilde]c'  parses to `Tilde[a,b,c]`. `\[Tilde]` has associated the Unicode character `\u223c`, while `\[RawTilde]` corresponds to the ASCII `~` (`\u007e`).

https://reference.wolfram.com/language/ref/character/Tilde.html
https://reference.wolfram.com/language/ref/character/RawTilde.html

### PR DESCRIPTION
**Pull request by mmatera**
Fixes #44

### CODE REVIEW

File: mathics_scanner/data/named-characters.yml [hunk 1]
```diff
@@ -4469,15 +4469,15 @@ Infinity:
...
+# See also RawTilde
 Infix:
   ascii: "~"
   amslatex: "\\textasciitilde"
   has-unicode-inverse: false
   is-letter-like: false
 # operator-name: Infix
   unicode-equivalent: "~"
-  wl-unicode-name: TILDE
+#  wl-unicode-name: RAWTILDE
```
Comment (rocky) [inline mathics_scanner/data/named-characters.yml:4480]: This comment is confusing. The one above says "See also RawTilde" so this isn't that. What purpose does the comment serve here?

**Comment by rocky** [comment]
LGTM - but please remove the garbage comment.

**Comment by rocky** [comment]
Actually, when I look the cumulative changes, I don't see that there is anything changed other than a "See also" comment. 

This habit of using a public shared git repository and Pull requests as a private sandbox is weird.  If locally you want to add print statements, debug statements side comments to yourself and so on, that is fine. But please when we put something in as a PR. Please go over everything and spend some time to describe the cumulative change, not your entire history of your thought process which may have reversed course a couple of times.

**Comment by mmatera** [comment]
> Actually, when I look the cumulative changes, I don't see that there is anything changed other than a "See also" comment, and removing the unicode equivalent association for Infix which is also known in WMA as Tilde.

Sorry for the many failed commits. I just checked it locally and seemed to work against my branch of mathics-core... but then I realized that it doesn't pass the local pytest.



> 
> This habit of using a public shared git repository and Pull requests as a private sandbox is weird. If locally you want to add print statements, debug statements side comments to yourself, and change your mind a dozen times about this, that is fine.
> 
> But please when we put something in as a PR, please go over everything and spend some time to describe the cumulative change and condensed final understanding, rather than show vague remnants of your entire history thought process which may have reversed course a couple of times.

Honestly, I thought that the change was easy, and then I realized that I didn't understand the logic behind this. Let me resume the changes:

* `\[Tilde]` is a character associated to the unicode `\u223c` instead of `~`
* `\[Tilde]` is an operator which is not `Infix` but `Tilde`, with precedence `290`.
* `\[Tilde]` has a `wl-unicode-name`  `TILDE OPERATOR`
* `~` is not a valid  ASCII representation for this operator (otherwise, the conversion get broken)
* `\[RawTilde]` is the (ASCII) character `~`, and it is associated to `Infix`, and is associated to the `wl-unicode` character `~` and
 `wl-unicode-name`  `TILDE`

With these changes, it seems that the conversion back and forward from (WL) Unicode and UTF8 / ASCII representations works here and in mathics-core.

**Comment by rocky** [comment]
> Sorry for the many failed commits. I just checked it locally and seemed to work against my branch of mathics-core... but then I realized that it doesn't pass the local pytest.
> 

Ok - run local pytest before pushing to github please. That is why I have been trying to keep local tests relatively fast. For me the testing takes 2 1/2 minutes.  

(This is why I am not in favor of running commit-time kinds of style checks, in the normal testing path and apparently since you don't seem to do this it is why you are happy to trash local testing with stuff that makes a pass over the entire code base but changes very infrequently, like summary check, or checking against a list of builtin function names)


> Honestly, I thought that the change was easy, and then I realized that I didn't understand the logic behind this. 

This too is a common pattern, so please make note of it and try to catch yourself. (I have this bad habit too) In your effort to be helpful you can rush off and do something without fully understanding the situation fully.

File: mathics_scanner/data/named-characters.yml [hunk 2]
```diff
@@ -4469,15 +4469,14 @@ Infinity:
...
 # Infix isn't really an operator
-# See also Tilde
+# See also RawTilde
 Infix:
   ascii: "~"
   amslatex: "\\textasciitilde"
   has-unicode-inverse: false
   is-letter-like: false
 # operator-name: Infix
   unicode-equivalent: "~"
```
Comment (rocky) [inline mathics_scanner/data/named-characters.yml:4479]: Something for another time. The comments at the top say that if unicode-equivalent is the same as ascii, please omit.

File: mathics_scanner/data/named-characters.yml [hunk 3]
```diff
@@ -6221,6 +6220,8 @@ RawTilde:
   has-unicode-inverse: false
   is-letter-like: false
   unicode-equivalent: '~'
+  wl-unicode: '~'
```
Comment (rocky) [inline mathics_scanner/data/named-characters.yml:6223]: Something for another time. The comments at the top say that if wl-equivalent is the same as ascii, please omit.

**Pull request merged**